applications/GeoMechanicsApplication/custom_processes/apply_constant_phreatic_multi_line_pressure_process.hpp

```
@@ -0,0 +1,346 @@
+        }
+
+        if (GravityDirection() == 0) {
+            mGravityDirectionCoordinates = XCoordinates();
+        }
+        else if (GravityDirection() == 1)
+        {
+            mGravityDirectionCoordinates = YCoordinates();
+        }
+        else if (GravityDirection() == 2)
+        {
+            mGravityDirectionCoordinates = ZCoordinates();
+        }
+
+        if (!std::is_sorted(HorizontalDirectionCoordinates().begin(), HorizontalDirectionCoordinates().end()))
+        {
+            KRATOS_ERROR << "The Horizontal Elements Coordinates are not ordered."
+                         << rParameters
+                         << std::endl;
+        }
+
+        mSpecificWeight = rParameters["specific_weight"].GetDouble();
+        if (rParameters.Has("pressure_tension_cut_off"))
+          mPressureTensionCutOff = rParameters["pressure_tension_cut_off"].GetDouble();
+        else
+          mPressureTensionCutOff = 0.0;
+
+        KRATOS_CATCH("")
+    }
+
+    ///------------------------------------------------------------------------------------
+
+    /// Destructor
+    ~ApplyConstantPhreaticMultiLinePressureProcess() override = default;
+
+    /// Assignment operator.
+    ApplyConstantPhreaticMultiLinePressureProcess& operator=(ApplyConstantPhreaticMultiLinePressureProcess const&) = delete;
+
+    /// Copy constructor.
+    ApplyConstantPhreaticMultiLinePressureProcess(ApplyConstantPhreaticMultiLinePressureProcess const&) = delete;
+
+///----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+    /// this function is designed for being called at the beginning of the computations
+    /// right after reading the model and the groups
+    void ExecuteInitialize() override
+    {
+        KRATOS_TRY
+
+        if (mrModelPart.NumberOfNodes() <= 0) {
+            return;
```

Do you think it might be better to throw an error here instead of just returning, or does it make sense to continue when the number of nodes is <= 0?

---

I added a throw here, let me know if you agree